### PR TITLE
Added ChainConfigResolver tests

### PR DESCRIPTION
**Pull request by mikadamczyk**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | N/A
| **Bug/Improvement**|no
| **New feature**    |no
| **Target version** | `master`
| **BC breaks**      |no
| **Tests pass**     | yes
| **Doc needed**     |no

**TODO**:
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+             ->with($relativeScopeParameter)
+             ->willReturn($expectedValue);
+
+        $this->assertSame($expectedValue, $this->getChainConfigResolver()->getParameter($paramName));
+    }
+
+    /**
+     * @dataProvider parameterProvider
+     */
+    public function testGetParameterSpecificScope(string $paramName, $expectedValue)
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php]:
```suggestion
    public function testGetParameterSpecificScope(string $paramName, $expectedValue): void
```

File: eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+        $this->assertSame(
+             $expectedValue,
+             $this->getChainConfigResolver()->getParameter($paramName, self::DEFAULT_NAMESPACE, self::FIRST_SA_NAME)
+         );
+    }
+
+    /**
+     * @dataProvider parameterProvider
+     */
+    public function testGetParameterGlobalScope(string $paramName, $expectedValue)
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php]:
```suggestion
    public function testGetParameterGlobalScope(string $paramName, $expectedValue): void
```

File: eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+    /**
+     * @dataProvider hasParameterProvider
+     */
+    public function testHasParameterNoNamespace(
+         bool $defaultMatch,
+         bool $groupMatch,
+         bool $scopeMatch,
+         bool $globalMatch,
+         bool $expectedResult
+     ) {
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php]:
```suggestion
     ): void {
```

File: eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    /**
+     * @dataProvider hasParameterProvider
+     */
+    public function testHasParameterWithNamespaceAndScope(
+         bool $defaultMatch,
+         bool $groupMatch,
+         bool $scopeMatch,
+         bool $globalMatch,
+         bool $expectedResult
+     ) {
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/Tests/DependencyInjection/Configuration/ConfigResolver/ChainConfigResolverTest.php]:
```suggestion
     ): void {
```

**Comment by mikadamczyk** [comment]
ping @alongosz

**Comment by alongosz** [comment]
(changes to tests only, no QA was required here)

**Pull request merged**